app/src/main/java/com/infomaniak/swisstransfer/upload/TransferUploader.kt

```diff
@@ -164,52 +182,87 @@ class TransferUploader(
         }
     }
 
-    private suspend fun uploadChunks(
+    private suspend fun syncFileChunksUploadStatuses(metadata: FileToUploadMetaData, totalChunks: Int) = coroutineScope {
+        for (chunkIndex in 0..<totalChunks) {
+            when (metadata.chunksUploadStatus[chunkIndex]) {
+                null, DefinitelyComplete -> continue // Leave as is.
+                StartedOrComplete -> launch {
+                    val exists: Boolean = uploadManager.doesChunkExist(
+                        remoteContainerUuid = destination.container.uuid,
+                        fileUUID = metadata.uuid,
+                        chunkIndex = chunkIndex,
+                        chunkSize = metadata.chunkConfig.fileChunkSize
+                    )
+                    metadata.chunksUploadStatus[chunkIndex] = if (exists) DefinitelyComplete else null
+                }
+            }
+        }
+    }
+
+    private suspend fun uploadFileInChunks(
+        metadata: FileToUploadMetaData,
         fileUUID: String,
-        chunkSize: Long,
         inputStream: BufferedInputStream,
-        parallelChunks: Int,
-        totalChunks: Int,
-        onUploadBytes: suspend (Long) -> Unit,
     ) {
-        val requestSemaphore = Semaphore(parallelChunks)
+
+        val chunkConfig = metadata.chunkConfig
+        val chunkSize = chunkConfig.fileChunkSize
+        val totalChunks = chunkConfig.totalChunks
+        val parallelChunks = chunkConfig.parallelChunks
+
```

Comment:
These definitions aren't really useful 
You could either make the function an extension of the metaData class (and move it in an extension File along with the other function up there), and you could also put the function as a with(chunkConfig) to already simplify this